File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0268_Missing_Number.java
package com.lanceolata.coding.algorithms.java.leetcode;

public class Question_0268_Missing_Number {
    public int missingNumber(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int n = nums.length;
        for (int i = 0; i < nums.length; i++) {
            n ^= i;
            n ^= nums[i];
        }
        return n;
    }

    public int missingNumber2(int[] nums) {
        if (nums == null || nums.length == 0) {
            return 0;
        }
        int n = 0;
        for (int i = 0; i < nums.length; i++) {
            n ^= i + 1;
            n ^= nums[i];
        }
        return n;
    }
}

File: algorithms/algorithms-cplusplus/leetcode/Question_0268_Missing_Number.cc
#include <vector>

using namespace std;

class Solution {
 public:
  int missingNumber(vector<int>& nums) {
    int len = nums.size();
    int sum = (0 + len) * (len + 1)/2;
    for (size_t i = 0; i < nums.size(); i++) {
      sum -= nums[i];
    }
    return sum;
  }

  int missingNumber2(vector<int>& nums) {
    int res = nums.size();
    for (size_t i = 0; i < nums.size(); i++) {
      res ^= i;
      res ^= nums[i];
    }
    return res;
  }
};
